### ISSUE TITLE
Ability to hide admin notices

### ISSUE THREAD

**Issue by stevejburge** [publishpress/PublishPress-Capabilities#135]
We could have an option to disable all admin notices for specific roles. Code here:
https://wordpress.stackexchange.com/questions/296056/hide-admin-notices-notifications-from-everyone-but-super-admin/385268

This could be a setting inside the "Roles" area.

**Comment by olatechpro** [comment]
Which menu are we adding this ?

@stevejburge

**Comment by olatechpro** [comment]
Nevermind mind my previous question.

I think this go to the new 'Admin Features' menu

@stevejburge

**Comment by stevejburge** [comment]
@olatechpro Yes, please https://github.com/publishpress/PublishPress-Capabilities/issues/118#issuecomment-922489647

**Comment by olatechpro** [comment]
Hi @stevejburge, i already implemented this changes but while testing, i noticed a big problem we may be causing with this feature and user may not know why.

If admin notice is disabled, all pages that send action notification via admin notice will be affected and not sure anyone will want that.

For example, after updating a WordPress taxonomy, creating/editing user or any action that require success or error message after saving, the success or error message won't show since the admin notices is disabled and people using the feature will not know since they're unaware admin notice is used to display all action notice and response message if in use.

What do you think, do we still leave this feature ?

**Comment by stevejburge** [comment]
@olatechpro Good catch. There's no way to distinguish the promotional messages from normal messages?

**Comment by olatechpro** [comment]
No, they all use same admin_notices hook.

Btwn, I noticed you changed this issue milestone, do you want me to remove it from current PR since I already commit the changes?

@stevejburge

**Comment by stevejburge** [comment]
@olatechpro yes please, if we don't think it's working correctly yet, or if we need another approach

**Comment by stevejburge** [comment]
I think the answer is probably a notices manager like one of these two plugins. Then you could hide that feature from some user roles.

![Image](https://github.com/user-attachments/assets/61d98139-5ee3-4b7b-b4a8-abcf6bcb6393)

Both those plugins have options like this:

![Image](https://github.com/user-attachments/assets/358edc47-6aca-4b5e-99ba-cfad6ed75b09)
![Image](https://github.com/user-attachments/assets/7c3d6692-d3a9-4b7b-a056-40fa2a47c08b)
